packages/data-context/src/sources/VersionsDataSource.ts

```diff
@@ -25,14 +25,20 @@ const NPM_CYPRESS_REGISTRY = 'https://registry.npmjs.org/cypress'
 export class VersionsDataSource {
   private _initialLaunch: boolean
   private _currentTestingType: TestingType | null
-  private _latestVersion: Promise<string>
-  private _npmMetadata: Promise<Record<string, string>>
 
   constructor (private ctx: DataContext) {
     this._initialLaunch = true
     this._currentTestingType = this.ctx.coreData.currentTestingType
-    this._latestVersion = this.getLatestVersion()
-    this._npmMetadata = this.getVersionMetadata()
+    this.#ensureData()
+  }
+
+  #ensureData () {
+    this.ctx.update((d) => {
+      d.versionData.latestVersion ??= this.getLatestVersion().catch((e) => undefined)
```

Comment:
What's the value of storing these on core data rather than on the data source?

Reply:
It was b/c the `VersionsDataSource` is cached on the data context, and since we keep the same data context globally, it was hard to reset the state for test.